examples/model_selection/plot_roc.py

```diff
@@ -71,39 +59,69 @@
                                  random_state=random_state))
 y_score = classifier.fit(X_train, y_train).decision_function(X_test)
 
+###############################################################################
+# Compute the AUC scores
+# ......................
+# The ROC area can be approximated by taking the average either weighted
+# uniformly or by the priori class distribution.
+from sklearn.metrics import roc_auc_score
+
+y_score_norm = y_score / y_score.sum(1, keepdims=True)
+unweighted_roc_auc_ovr = roc_auc_score(y_test, y_score_norm, multiclass="ovr")
+weighted_roc_auc_ovr = roc_auc_score(
+      y_test, y_score_norm, multiclass="ovr", average="weighted")
+print("One-vs-Rest ROC AUC scores: {0} (unweighted), {1} (weighted)".format(
+       unweighted_roc_auc_ovr, weighted_roc_auc_ovr))
+
+###############################################################################
+# Plotting the ROC curve for virginica
+# ....................................
+import matplotlib.pyplot as plt
+from sklearn.preprocessing import label_binarize
+from sklearn.metrics import roc_curve, auc
```

Comment:
I think all of the imports should be at the top of the file

Reply:
When looking at the rendered html: https://45452-843222-gh.circle-artifacts.com/0/doc/auto_examples/model_selection/plot_roc.html#sphx-glr-auto-examples-model-selection-plot-roc-py, the imports are place right before they are used in the code block. I think this makes it easier to read, without a big block of imports in the beginning of the file.

Given that, I do not mind placing them no top, since the functions/classes are linked to their documentation.

Reply:
I agree it makes the code slightly better when looking at the html. OTOH at first it made me believe that each section is "stand-alone", i.e. that you could run it without having run the previous sections which is not the case.

Let's see what others say.